pandas/core/reshape/pivot.py

```diff
@@ -424,17 +435,26 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
-def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+def pivot(
+    data: "DataFrame",
+    index: Optional[Union[Label, Sequence[Label]]] = None,
+    columns: Optional[Union[Label, Sequence[Label]]] = None,
+    values: Optional[Union[Label, Sequence[Label]]] = None,
+) -> "DataFrame":
     if columns is None:
         raise TypeError("pivot() missing 1 required argument: 'columns'")
-    columns = columns if is_list_like(columns) else [columns]
+
+    if is_list_like(columns):
+        columns = com.convert_to_list_like(columns)
+    else:
+        columns = [columns]
 
     if values is None:
-        cols: List[str] = []
+        cols: List[Label] = []
         if index is None:
             pass
         elif is_list_like(index):
-            cols = list(index)
+            cols = com.convert_to_list_like(index)
```

Comment:
you shouldn't be able to get rid of the is_list_like now?

Reply:
good point!